OpenRA.Mods.Common/Effects/AreaBeam.cs

```diff
@@ -0,0 +1,216 @@
+#region Copyright & License Information
+/*
+ * Copyright 2007-2015 The OpenRA Developers (see AUTHORS)
+ * This file is part of OpenRA, which is free software. It is made
+ * available to you under the terms of the GNU General Public License
+ * as published by the Free Software Foundation. For more information,
+ * see COPYING.
+ */
+#endregion
+
+using System;
+using System.Collections.Generic;
+using System.Drawing;
+using System.Linq;
+using OpenRA.Effects;
+using OpenRA.GameRules;
+using OpenRA.Graphics;
+using OpenRA.Mods.Common.Graphics;
+using OpenRA.Mods.Common.Traits;
+using OpenRA.Traits;
+
+namespace OpenRA.Mods.Common.Effects
+{
+	public class AreaBeamInfo : IProjectileInfo
+	{
+		[Desc("Projectile speed in WDist / tick, two values indicate a randomly picked velocity per beam.")]
+		public readonly WDist[] Speed = { new WDist(17) };
+
+		[Desc("The maximum duration (in ticks) of each beam burst.")]
+		public readonly int Duration = 10;
+
+		[Desc("The number of ticks between the beam causing warhead impacts in its area of effect.")]
+		public readonly int DamageInterval = 3;
+
+		[Desc("The width of the beam.")]
+		public readonly WDist Width = new WDist(512);
+
+		[Desc("How far beyond the target the projectile keeps on travelling.")]
+		public readonly WDist BeyondTargetRange = new WDist(128);
+
+		[Desc("Damage modifier applied at each range step.")]
+		public readonly int[] Falloff = { 100, 100 };
+
+		[Desc("Ranges at which each Falloff step is defined.")]
+		public readonly WDist[] Range = { WDist.Zero, new WDist(int.MaxValue) };
+
+		[Desc("Maximum offset at the maximum range.")]
+		public readonly WDist Inaccuracy = WDist.Zero;
+
+		[Desc("Can this projectile be blocked when hitting actors with an IBlocksProjectiles trait.")]
+		public readonly bool Blockable = false;
+
+		[Desc("Extra search radius beyond beam width. Required to ensure affecting actors with large health radius.")]
+		public readonly WDist TargetExtraSearchRadius = new WDist(2048);
+
+		[Desc("Should the beam be visuall rendered? False = Beam is invisible.")]
+		public readonly bool RenderBeam = true;
+
+		[Desc("Color of the beam.")]
+		public readonly Color Color = Color.Red;
+
+		[Desc("Beam color is the player's color.")]
+		public readonly bool UsePlayerColor = false;
+
+		public IEffect Create(ProjectileArgs args)
+		{
+			var c = UsePlayerColor ? args.SourceActor.Owner.Color.RGB : Color;
+			return new AreaBeam(this, args, c);
+		}
+	}
+
+	public class AreaBeam : IEffect, ISync
+	{
+		readonly AreaBeamInfo info;
+		readonly ProjectileArgs args;
+		readonly AttackBase actorAttackBase;
+		readonly Color color;
+		readonly WDist speed;
+
+		[Sync] WPos headPos;
+		[Sync] WPos tailPos;
+		[Sync] WPos target;
+		int length;
+		int facing;
+		int headTicks;
+		int tailTicks;
+		bool headTravelling = true;
+		bool tailTravelling;
+
+		bool BeamComplete { get { return !headTravelling && headTicks >= length &&
```

Comment:
`IsBeamComplete`, please.
Same for the other `bool` fields - `isHeadTravelling` and `isTailTravelling`, respectively.